internal/bminventory/inventory.go

```diff
@@ -4011,18 +3993,76 @@ func (b *bareMetalInventory) RegisterInfraEnvInternal(
 		return nil, common.NewApiError(http.StatusBadRequest, err)
 	}
 
-	if err = b.db.Create(&infraEnv).Error; err != nil {
+	if err = tx.Create(&infraEnv).Error; err != nil {
+		log.WithError(err).Error("failed to create infraenv")
+		return nil, common.NewApiError(http.StatusInternalServerError, err)
+	}
+
+	if err = tx.Commit().Error; err != nil {
+		log.WithError(err).Error("failed to commit transaction registering infraenv")
 		return nil, common.NewApiError(http.StatusInternalServerError, err)
 	}
 
+	success = true
 	if err = b.GenerateInfraEnvISOInternal(ctx, &infraEnv); err != nil {
 		return nil, err
 	}
 
-	success = true
 	return b.GetInfraEnvInternal(ctx, installer.GetInfraEnvParams{InfraEnvID: *infraEnv.ID})
 }
 
+func (b *bareMetalInventory) validateInfraEnvCreateParams(ctx context.Context, params installer.RegisterInfraEnvParams, cluster *common.Cluster) error {
+	var err error
+	if cluster != nil && cluster.CPUArchitecture != "" && cluster.CPUArchitecture != params.InfraenvCreateParams.CPUArchitecture {
+		err = errors.Errorf("Specified CPU architecture doesn't match the cluster (%s)",
+			cluster.CPUArchitecture)
+		return err
+	}
+
+	if err = auth.ValidateAccessToCPUArchitecture(ctx, b.authzHandler, params.InfraenvCreateParams.CPUArchitecture); err != nil {
+		b.log.WithError(err).Errorf("error validaitng access to cpu arch")
+		return err
+	}
+
+	if params.InfraenvCreateParams.Proxy != nil {
+		if err = validateProxySettings(params.InfraenvCreateParams.Proxy.HTTPProxy,
+			params.InfraenvCreateParams.Proxy.HTTPSProxy,
+			params.InfraenvCreateParams.Proxy.NoProxy, &params.InfraenvCreateParams.OpenshiftVersion); err != nil {
+			b.log.WithError(err).Errorf("error validaitngproxy ")
+
+			return err
+		}
+	}
+
+	ntpSource := swag.StringValue(params.InfraenvCreateParams.AdditionalNtpSources)
+	if ntpSource != b.Config.DefaultNTPSource && !pkgvalidations.ValidateAdditionalNTPSource(ntpSource) {
+		err = errors.Errorf("Invalid NTP source: %s", ntpSource)
+		return err
+	}
+
+	if params.InfraenvCreateParams.SSHAuthorizedKey != nil && *params.InfraenvCreateParams.SSHAuthorizedKey != "" {
+		if err = validations.ValidateSSHPublicKey(*params.InfraenvCreateParams.SSHAuthorizedKey); err != nil {
+
+			err = errors.Errorf("SSH key is not valid")
+			b.log.WithError(err).Errorf("error validating ssh key")
```

Comment:
Why add a log here? We didn't log before, right?

Reply:
Oops thank you! This was a debug log I accidentally left in